keychain.go

```diff
@@ -237,6 +237,19 @@ func AddItem(item Item) error {
 	return err
 }
 
+// UpdateItem updates a Item
+func UpdateItem(item Item) error {
+	cfDict, err := ConvertMapToCFDictionary(item.attr)
+	if err != nil {
+		return err
+	}
+	defer Release(C.CFTypeRef(cfDict))
+
+	errCode := C.SecItemUpdate(cfDict, cfDict)
```

Comment:
Is it awkward to pass in the same dict as both args? or is that the normal way to update?